src/test/java/org/broadinstitute/hellbender/utils/interval/IntervalUtilsUnitTest.java

```diff
@@ -1109,4 +1110,23 @@ public void testGenomeLocsFromLocatables(final GenomeLocParser parser, final Lis
         };
     }
 
+    @Test(dataProvider = "overlapData")
+    public void testOverlap(final Locatable l, final Locatable r, final boolean expected) {
+        final boolean actual = IntervalUtils.overlaps(l, r);
+        Assert.assertEquals(actual, expected);
+    }
+
+    @DataProvider(name = "overlapData")
+    public Object[][] overlapData() {
+        return new Object[][]{
+                {new SimpleInterval("1", 1, 2), new SimpleInterval("1", 2, 3), true},
+                {new SimpleInterval("1", 1, 2), new SimpleInterval("1", 3, 4), false},
+                {new SimpleInterval("1", 10, 20), new SimpleInterval("2", 10, 20), false},
+                {new SimpleFeature(null, 10, 20), new SimpleFeature(null, 10, 20), false},
+                {new SimpleFeature("1", 10, 20), new SimpleInterval("1", 10, 20), true},
+                {new SimpleInterval("1", 10, 20), new SimpleFeature("1", 10, 20), true},
+                {new SimpleFeature(null, 10, 20), new SimpleFeature("1", 10, 20), false},
+                {new SimpleFeature("1", 10, 20), new SimpleFeature(null, 10, 20), false},
+        };
```

Comment:
Re-use the more exhaustive set of interval overlap test data from `SimpleIntervalUnitTest`